superset/views/alerts.py

```diff
@@ -23,17 +24,62 @@
 from wtforms import BooleanField, Form, StringField
 
 from superset.constants import RouteMethod
-from superset.models.alerts import Alert, AlertLog
+from superset.models.alerts import (
+    Alert,
+    AlertLog,
+    AlertValidatorType,
+    SQLObservation,
+    SQLObserver,
+    Validator,
+)
 from superset.models.schedules import ScheduleType
 from superset.tasks.schedules import schedule_alert_query
+from superset.utils import core as utils
 from superset.utils.core import get_email_address_str, markdown
 
 from ..exceptions import SupersetException
+from ..sql_parse import ParsedQuery
 from .base import SupersetModelView
 
 # TODO: access control rules for this module
 
 
+def test_observer_sql(item: "SQLObserverInlineView") -> None:
+    try:
+        parsed_query = ParsedQuery(item.sql)
+        sql = parsed_query.stripped()
+        df = item.database.get_df(sql)
+
+        if not df.empty:
+            value = df.to_records()[0][1]
+
+            # Check that the SQL result can be read as a float
+            if value:
+                float(value)
+
+    except ValueError:
+        raise SupersetException("Error: SQL query returned non-number result")
+    except Exception as ex:  # pylint: disable=broad-except
+        raise SupersetException(f"Observer raised exception: {ex}")
+
+
+def check_validator_config(item: "ValidatorInlineView") -> None:
+    config = json.loads(item.config)
+
+    if item.validator_type == AlertValidatorType.gte_threshold and not config.get(
+        "gte_threshold"
+    ):
+        raise SupersetException(
+            "Error: Greater Than or Equal To Validator needs a specified threshold"
```

Comment:
in exception - specify and example, this way user will have an actionable insight